### PR TITLE
utils: Allow passing NULL for error

### PR DESCRIPTION
**Pull request by vojtechtrefny**
Just the utils library for now, I really look forward to doing this for every plugin (it should at least be easier once all the exec functions are "safe").

Related: #527

### CODE REVIEW

**Review by tbzatek — APPROVED** [review]
Didn't check excessively deep, looks fine though.

Thanks for making this finally happen!

File: src/utils/dbus.c [hunk 1]
```diff
@@ -59,7 +59,7 @@ gboolean bd_utils_dbus_service_available (GDBusConnection *connection, GBusType
     else {
         bus = g_bus_get_sync (bus_type, NULL, error);
         if (!bus) {
-            g_critical ("Failed to get system bus: %s\n", (*error)->message);
```
[the remark this answers is not in the document]
Comment (vojtechtrefny) [reply src/utils/dbus.c:62]: libblockdev is a library, we should not print to stdout/stderr. If you think we should log this, we can do that, but in this case we set the error and that should be enough -- if user wants to ignore the error, it's his problem, not ours

[the remark this answers is not in the document]
Comment (tbzatek) [reply src/utils/dbus.c:62]: Ah right. What about at least logging through the `libblockdev` logging channels if they're initialized at that point? Things with critical severity might deserve that.

[the remark this answers is not in the document]
Comment (vojtechtrefny) [reply src/utils/dbus.c:62]: And is this really a critical error? User asks if service X is available and gets "no, error happened when I tried to check", that's not end of the world.

[the remark this answers is not in the document]
Comment (tbzatek) [reply src/utils/dbus.c:62]: Depends. Error reaching the system bus is generally critical as other things would likely fail the same way. Might not be an issue in restricted environments such as embedded systems or the initramfs. Of course it's the caller responsibility to catch and process the error further.

Anyway, the change is probably good enough as it is, just wanted to point out that critical severity might deserve additional logging. It was done that way until now after all. Not a hard argument though, we're breaking the API anyway.

**Pull request merged**